dust_attenuation/shapes.py

```diff
@@ -0,0 +1,391 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+import astropy.units as u
+
+from .baseclasses import BaseAttAvModel
+from .helpers import _test_valid_x_range
+
+from .averages import C00, Leitherer02
+from astropy.modeling import Parameter, InputParameterError
+
+__all__ = ['Noll09']
+
+x_range_Noll09 = [0.097, 2.2]
+
+
+class Noll09(BaseAttAvModel):
+    """
+    Attenuation curve using a modified version of the Calzetti law
+    allowing for a varying UV slope and the presence of a UV bump.
+
+    Parameters
+    ----------
+    x0: float
+        Central wavelength of the UV bump (in microns).
+
+    gamma: float
+        Width (FWHM) of thhe UV bump (in microns).
+
+    ampl: float
+        Amplitude of the UV bump.
+
+    slope: float
+        Slope of the power law.
+
+    Av: float
+        attenuation in V band.
+
+    Raises
+    ------
+    InputParameterError
+       Input Av values outside of defined range
+
+    Notes
+    -----
+
+    The original formalism is from Noll et al, A&A 507, 1793–1813 (2009).
+    However the UV bump was added before applying the power law correction.
+    In this class we follow the modified version of Salim et al 2018 (The
+    Astrophysical Journal, Volume 859, Issue 1, article id. 11, 17 pp.)
+     where the UV bump is added after the power law correction.
```

Comment:
How significant is this change?  Will this confuse some users that use the original Noll 09 version versus the Salim 18 version?  If so, then I think we should implement both models.  One idea for this package is to provide both the most up-to-date and historical curves to allow users to pick, and be able to show the differences explicitly.  And educate them why one is better than the other.  :-)

Reply:
The change is sufficiently significant to make 2 distinct models ;)
In the original formulation of Noll09, the UV bump is affected by the power-law:
k_Noll09(lambda) = [k_C00(lambda) + UV_bump] * (lambda / 0.55)^slope 

And in some work, also referring to Noll09, it is not:
k_Noll09(lambda) = k_C00(lambda) * (lambda / 0.55)^slope + UV_bump

I prefer the second formulation, but I agree with you both should be available.
I need to find out the first paper using the second formulation. Or maybe N09 and N09mod?

Reply:
It would be better to find the first paper that made the modification to N09 and name it appropriately.  Nice to give credit where credit is due.

Reply:
I have checked in the papers citing Noll 2009 and apparently the paper of Salim, Boquien and Lee 2018 is the first mention to this modification in the N09 formalism. So there are two classes now: N09 and SBL18